evm_loader/cli/src/api_server/handlers/emulate.rs

```diff
@@ -1,42 +1,28 @@
-use tide::{Request, Result};
+use axum::{http::StatusCode, Json};
 
 use crate::{
-    api_server::state::State, commands::emulate as EmulateCommand, context,
-    types::request_models::EmulateRequestModel,
+    commands::emulate as EmulateCommand, context, types::request_models::EmulateRequestModel,
+    NeonApiState,
 };
 
-use super::{parse_emulation_params, process_result};
+use super::{parse_emulation_params, process_error, process_result};
 
 #[allow(clippy::unused_async)]
-pub async fn emulate(mut req: Request<State>) -> Result<serde_json::Value> {
-    let emulate_request: EmulateRequestModel = req.body_json().await.map_err(|e| {
-        tide::Error::from_str(
-            400,
-            format!(
-                "Error on parsing transaction parameters request: {:?}",
-                e.to_string()
-            ),
-        )
-    })?;
-
-    let state = req.state();
-
+pub async fn emulate(
+    axum::extract::State(state): axum::extract::State<NeonApiState>,
+    Json(emulate_request): Json<EmulateRequestModel>,
+) -> (StatusCode, Json<serde_json::Value>) {
     let tx = emulate_request.tx_params.into();
 
-    let signer = context::build_singer(&state.config).map_err(|e| {
-        tide::Error::from_str(
-            400,
-            format!("Error on creating singer: {:?}", e.to_string()),
-        )
-    })?;
-
-    let rpc_client =
-        context::build_rpc_client(&state.config, emulate_request.slot).map_err(|e| {
-            tide::Error::from_str(
-                400,
-                format!("Error on creating rpc client: {:?}", e.to_string()),
-            )
-        })?;
+    let signer = match context::build_singer(&state.config) {
```

Comment:
I think the name should be `build_signer` instead of `build_singer`. There are also other places apart from this one with the `singer` vs `signer` typo.

Reply:
fixed

Reply:
I think this should not have happened: `TxParametersParsignerror`